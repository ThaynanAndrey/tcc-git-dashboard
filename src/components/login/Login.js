import React, { Component } from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { login, isAuthenticated } from '../../store/actions/authAction';
import { ToastContainer, toast } from 'react-toastify';

const cardStyles = {
    textAlign: "center",
    width: "38vw",
    margin: "0 auto",
    flex: "1 1 auto",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center"
};

const buttonStyle = {
    width: "65%"
};

/**
 * Component with the function of presenting Login Screen.
 * 
 * @author Thaynan Nunes
 */
class Login extends Component {
  constructor(props) {
    super(props);
    
    this.login = this.login.bind(this);
  }

  /**
   * Login user.
   */
  async login() {
    await this.props.login();

    if(isAuthenticated()) {
        this.props.history.push('/');
    } else {
        toast.error("Error ao tentar logar! Tente Novamente", {
            position: "top-right",
            autoClose: 4500,
            hideProgressBar: true,
            closeOnClick: true,
            pauseOnHover: true,
            draggable: true
        });
    }
  }

  render() {
    return (
        <div style={cardStyles}>
            <ToastContainer />
            
            <div className="card blue-grey darken-2">
                <div className="card-content white-text">
                    <span className="card-title">
                        <h4>
                            Git Dashboard
                        </h4>    
                    </span>
                    <h6>
                        Autentique-se com sua conta do GitHub para poder
                        ter acesso aos seus Reposit√≥rios e Pull Requests.
                    </h6>
                </div>
                <div className="card-action">
                    <button className="waves-effect waves-light btn blue lighten-1"
                        style={buttonStyle}
                        onClick={() => this.login()}>
                        <FontAwesomeIcon icon={['fab', 'github']} />
                        &nbsp;
                        Login com GitHub
                    </button>
                </div>
            </div>
        </div>
    );
  }
}

const mapStateToProps = (state) => ({
    user: state.auth.user,
    error: state.auth.errorLogin
});

const mapDispatchToProps = (dispatch) => ({
    login: () => dispatch(login())
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);