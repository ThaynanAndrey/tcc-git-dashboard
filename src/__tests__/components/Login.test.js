import React from 'react';
import { shallow, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import Login from '../../components/login/Login';

describe('Test Login', () => {
  it('renders correctly a Login\'s component', () => {
    const tree = shallow(
      <Login/>
    );

    expect(toJson(tree)).toMatchSnapshot();
  });
})