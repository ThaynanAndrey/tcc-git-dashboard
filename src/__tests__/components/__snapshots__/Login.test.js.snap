// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login renders correctly a Login's component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
